<launch>
  <arg name="input_route" default="input_route"/>
  <arg name="input_occupancy_grid" default="input_occupancy_grid"/>
  <arg name="input_scenario" default="input_scenario"/>
  <arg name="input_twist" default="/localization/twist" />
  <arg name="output_trajectory" default="output_trajectory"/>
  <arg name="is_completed" default="is_completed"/>

  <node pkg="freespace_planner" exec="freespace_planner" name="freespace_planner" output="screen">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>
    <remap from="~/input/route" to="$(var input_route)" />
    <remap from="~/input/occupancy_grid" to="$(var input_occupancy_grid)" />
    <remap from="~/input/scenario" to="$(var input_scenario)" />
    <remap from="~/input/twist" to="$(var input_twist)" />
    <remap from="~/output/trajectory" to="$(var output_trajectory)" />
    <remap from="is_completed" to="$(var is_completed)" />

    <!-- Node configurations -->
    <param name="waypoints_velocity" value="5.0" />
    <param name="update_rate" value="10.0" />
    <param name="th_arrived_distance_m" value="1.0" />
    <param name="th_stopped_time_sec" value="1.0" />
    <param name="th_stopped_velocity_mps" value="0.01" />
    <param name="th_course_out_distance_m" value="1.0" />
    <param name="replan_when_obstacle_found" value="true" />
    <param name="replan_when_course_out" value="true" />

    <!-- A* search configurations -->
    <param name="use_back" value="true" />
    <param name="only_behind_solutions" value="false" />
    <param name="time_limit" value="30000.0" />
    <param name="robot_length" value="4.5" />
    <param name="robot_width" value="1.75" />
    <param name="robot_base2back" value="1.0" />
    <param name="minimum_turning_radius" value="9.0" />
    <param name="theta_size" value="144" />
    <param name="angle_goal_range" value="6.0" />
    <param name="curve_weight" value="1.2" />
    <param name="reverse_weight" value="2.0" />
    <param name="lateral_goal_range" value="0.5" />
    <param name="longitudinal_goal_range" value="2.0" />
    <param name="obstacle_threshold" value="100" />
    <param name="distance_heuristic_weight" value="1.0" />
  </node>
</launch>
